### DISCUSSION TITLE
Clarification on how the _bulk_get endpoint is supposed to work with regards to _deleted documents

### DISCUSSION THREAD

**Discussion by broomfn** [apache/couchdb#3849] [Q&A]
For reference please see: https://github.com/matteobortolazzo/couchdb-net/issues/154

In summary, if I use Thunder Client/Postman and pass a deleted document id such as below to the _bulk_get CouchDB endpoint:

```
{
    "docs": [
        {
            "id": "rb-ct002110:017d33dd6cf4476f9e6e8604fd436c73"
        }
    ]
}
```

I get the following result:

```
{
  "results": [
    {
      "id": "rb-ct002110:017d33dd6cf4476f9e6e8604fd436c73",
      "docs": [
        {
          "ok": {
            "_id": "rb-ct002110:017d33dd6cf4476f9e6e8604fd436c73",
            "_rev": "2-8006f354c92f25ef9cf4aa630fac35fc",
            "_deleted": true
          }
        }
      ]
    }
  ]
}
```

The documentation for this function, doesn't mention the _deleted flag, so it's not clear whether this is expected behaviour or a bug:

https://docs.couchdb.org/en/stable/api/database/bulk-api.html?highlight=_bulk_get#db-bulk-get

The best information I can find on the _deleted flag is below and in regard to the GET /{db}/{docid} endpoint

https://docs.couchdb.org/en/stable/api/document/common.html#get--db-docid
https://docs.couchdb.org/en/stable/api/document/common.html#retrieving-deleted-documents

And this seems to suggest that to retrieve a deleted document you need to provide the rev as well?

Clarification on how the _bulk_get endpoint is suppose to work would be most welcome.

Many thanks

Neil

**Comment by nickva** [comment]
There was a similar issue reported a while back https://github.com/apache/couchdb/issues/3608. We had only verified that it behaved similarly to PouchDB
